src/robert/getAncestors.ts

```diff
@@ -0,0 +1,29 @@
+import { Tree } from '../types';
+
+export interface GetAnswersRecursiveResult { ancestors: string[]; found: boolean; }
+
+export function getAncestorsRecursive(tree: Tree, nodeName: string, ancestorsFound: string[]): GetAnswersRecursiveResult {
+    const { root } = tree
+    if (nodeName === root) {
+        ancestorsFound.push(root);
+        return { ancestors: ancestorsFound, found: true };
+    }
+
+    if (tree.children.length === 0) {
+        return { ancestors: ancestorsFound, found: false };
+    }
+
+    let ancestors = { ...ancestorsFound };
+    for (const child of tree.children) {
+        const results = getAncestorsRecursive(child, nodeName, ancestorsFound);
+        if (results.found) {
+            ancestors: results.ancestors.push(root)
```

Comment:
@YizYah does this work using an : in place of =

Reply:
Sorry, I'm confused. Do you mean in line 18? `results` is being declared there as a constant. So wouldn't I have to use `=`?  It does seem to work.